app/src/main/java/com/fcaputo/parkingapp/mvp/view/SettingsView.kt

```diff
@@ -0,0 +1,76 @@
+package com.fcaputo.parkingapp.mvp.view
+
+import android.app.Activity
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import com.fcaputo.parkingapp.R
+import com.fcaputo.parkingapp.databinding.ActivitySettingsBinding
+import com.fcaputo.parkingapp.mvp.contract.SettingsContract
+import com.fcaputo.parkingapp.mvp.view.base.ActivityView
+import com.fcaputo.parkingapp.utils.Constants
+import com.fcaputo.parkingapp.utils.validation.ValidationErrorType
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+class SettingsView(activity: Activity) : ActivityView(activity),SettingsContract.View {
+    private var binding: ActivitySettingsBinding = ActivitySettingsBinding.inflate(activity.layoutInflater)
+
+    init {
+        activity.setContentView(binding.root)
+    }
+
+    override fun onSaveButton(onClick: () -> Unit) {
+        binding.saveButton.setOnClickListener{
+            if(fieldsAreComplete()) {
+                onClick()
+            } else {
+                showErrorMessage(ValidationErrorType.INCOMPLETE_FIELD)
+            }
+        }
+    }
+
+    private fun fieldsAreComplete(): Boolean = listOf(binding.tInputParkingSize).all{ it.editText!!.text.isNotEmpty() }
```

Comment:
do not use !!, use ?: instead to avoid possibles crashes

Reply:
In fact, this method should be in the presenter, not here

Reply:
> do not use !!, use ?: instead to avoid possibles crashes

Done. 
I just thought that it was clearer to read, since I know for sure that the EditText element is present (cause it's there in the layout). But I know it's not recommended.


> In fact, this method should be in the presenter, not here

How could I achieve this in the presenter? since it must not know the list of UI elements (text inputs in this case)

Reply:
the validation to know if the fields are not empty should be in the presenter or model, depending the implementation, but not here in the view. From here you will never be able to test that

Reply:
Got it. Done.